mesonbuild/modules/cmake.py

```diff
@@ -256,31 +268,35 @@ def detect_cmake(self, state):
         self.cmake_detected = True
         return True
 
-    @permittedKwargs({'version', 'name', 'compatibility', 'install_dir'})
-    def write_basic_package_version_file(self, state, _args, kwargs):
-        version = kwargs.get('version', None)
-        if not isinstance(version, str):
-            raise mesonlib.MesonException('Version must be specified.')
+    # TODO: add release snippet
+    @typed_kwargs(
+        'cmake.write_basic_package_version_file',
+        KwargInfo('arch_independent', bool, default=False, since='0.62.0'),
+        KwargInfo('compatibility', str, default='AnyNewerVersion'),
```

Comment:
```suggestion
        KwargInfo('compatibility', str, default='AnyNewerVersion', validator=somefunc),
```

For the validator function `somefunc`, many parts of meson use one-off lambdas to define a completely custom function. But "must be one of a specific set of strings" is a common thing to need, so here we can use:
```python
validator=in_set_validator(set(COMPATIBILITIES))
```
(This function comes from `mesonbuild.interpreter.type_checking` and returns an inner function for you.)